TF-33216: Improve Terraform Enterprise Replicated-to-Docker migration guide (#1805)
hashicorp/web-unified-docs: content/terraform-enterprise/1.0.x/docs/enterprise/deploy/replicated-migration.mdx

Author: <img width="1520" height="138" alt="image" src="https://github.com/user-attachments/assets/b458c539-5400-4aaf-a1e4-55a1a664d4cc" />

   - For mounted disk installations, refer to [Backup a Mounted Disk Deployment](/terraform/tutorials/recommended-patterns/pattern-backups#backup-a-mounted-disk-deployment) in the Terraform Enterprise backup tutorial for recommended patterns.
   - For external services or active active installations, refer to [Object Store](/terraform/tutorials/recommended-patterns/pattern-backups#object-store) and [Database](/terraform/tutorials/recommended-patterns/pattern-backups#database) in the Terraform Enterprise backup tutorial for recommended patterns.

   In both cases, we recommend backing up the environment variables of the Terraform Enterprise Docker container so that you can refer to them when configuring your target runtime.

   Commands throughout this guide refer to `terraform-enterprise` as the container name, which is the default for Replicated installations. If your container has a different name, adjust the commands accordingly. Use the following command to find your container name:

   ```shell-session
   $ sudo docker ps | grep terraform-enterprise
   ```

>    Back up the environment variables:

Reviewer: ````suggestion
Commands throughout this guide refer to `terraform-enterprise` as the container name, which is the default for Replicated installations. If your container has a different name, adjust the commands accordingly. Use the following command to find your container name:

```shell-session
$ sudo docker ps | grep terraform-enterprise
```

Back up the environment variables:
````
Just to fix the alignment

   ```shell-session
   $ sudo docker exec terraform-enterprise env | sudo tee env.txt > /dev/null

>    ```

Reviewer: ````suggestion
$ sudo docker exec terraform-enterprise env | sudo tee env.txt > /dev/null
```
````

   In both cases, we recommend backing up the environment variables of the Terraform Enterprise Docker container so that you can refer to them when configuring your target runtime.

   Commands throughout this guide refer to `terraform-enterprise` as the container name, which is the default for Replicated installations. If your container has a different name, adjust the commands accordingly. Use the following command to find your container name:

>    `sudo docker ps | grep terraform-enterprise`

Reviewer: ````suggestion
```shell-session
$ sudo docker ps | grep terraform-enterprise
```
````
For consistency with other ways we format commands

      ```shell-session
      $ tfe-admin health-check
      ```

>       -> **Note:** If `tfe-admin health-check` fails with `sh: /root/ptfe-health-check: not found`, try running `replicated admin health-check` instead. If neither command works, refer to [tfe-admin health-check command fails](https://support.hashicorp.com/hc/en-us/articles/6053728286483-tfe-admin-health-check-command-fails-with-sh-root-ptfe-health-check-not-found) for additional troubleshooting. This issue is resolved in v202205-1 and later. **We recommend upgrading to the latest version if possible.**

Author: <img width="2082" height="827" alt="image" src="https://github.com/user-attachments/assets/4066346d-a824-4504-ba21-a331b112665c" />
Before (Left) vs After (Right)

   1. Do a `terraform plan` and a `terraform apply`.

### Prepare TLS certificates

> -> **Note**: Have your TLS certificates ready before stopping Replicated. This ensures a smooth migration without delays.

Reviewer: ```suggestion
Creating your TLS certificates before stopping Replicated ensures a smooth migration without delays.
```
It wasn't clear that the note was intended to reinforce the need for this step. I initially read this as saying that you should have TLS certificates before you prepare TLS certificates per this step.

1. Create the directory structure for Terraform Enterprise:
   ```shell-session
   $ sudo mkdir -p /etc/terraform-enterprise/certs
   ```

1. Copy your existing Replicated TLS certificates to the staging location:

   Retrieve the certificate paths from the container:
   ```shell-session
   $ sudo docker exec terraform-enterprise tfectl app config --unredacted | jq '{cert: .tls.cert_file, key: .tls.key_file, bundle: .tls.ca_bundle_file}'
   ```
   

>    The output will show the certificate paths in the container. Depending on your setup, the file paths may differ from the following example:

Reviewer: ```suggestion
   The output shows the certificate paths in the container. Depending on your setup, the file paths may differ from the following example:
```

1. Log in to the Terraform Enterprise container image registry.

   ```shell-session
   $ cat <PATH_TO_HASHICORP_LICENSE_FILE> | sudo docker login --username terraform images.releases.hashicorp.com --password-stdin
   ```

1. Pull the Terraform Enterprise image from the registry.  Refer to [Terraform Enterprise Releases](https://developer.hashicorp.com/terraform/enterprise/releases) for a list of versions and replace `<vYYYYMM-#>` with the version you wish to install; `latest` is not a valid tag.  

   ```shell-session
   $ sudo docker pull images.releases.hashicorp.com/hashicorp/terraform-enterprise:<vYYYYMM-#>
   ```

1. Create a new `systemd` service for Terraform Enterprise by creating a `/etc/systemd/system/terraform-enterprise.service` file with [the contents on the Docker Installation guide](/terraform/enterprise/deploy/docker). First, verify your Docker installation paths:

   ```shell-session
   $ which docker
   ```

>    Use the following command to create the systemd service file. The example below assumes Docker is installed at `/usr/bin/docker`. Adjust the paths in the configuration if your Docker installation differs:

Author: <img width="1511" height="101" alt="image" src="https://github.com/user-attachments/assets/37a062de-109a-4944-b6e4-50d1609bd7da" />

   ```shell-session
   $ replicatedctl app stop
   ```

>    ~> **Reminder:** If you have not already backed up your data, follow the steps in [Back up your data tier](#step-1-backup-your-data-tier) before proceeding, as the migration occurs in the following steps.

Author: <img width="2067" height="342" alt="Image" src="https://github.com/user-attachments/assets/2e8f8723-422a-4fc3-b648-6e6b12bbb0a1" />
Change applied. Image just shows that I reverted back to the original instructions for Step 6

1. Shut down Replicated.

   ```shell-session
   $ sudo systemctl disable --now replicated replicated-ui replicated-operator
   ```

1. Next, stop and remove the unnecessary Replicated containers.
   ```shell-session
   $ sudo docker stop replicated-premkit
   ```
   ```shell-session
   $ sudo docker stop replicated-statsd
   ```
   ```shell-session
   $ sudo docker rm -f replicated replicated-ui replicated-operator replicated-premkit replicated-statsd retraced-api retraced-processor retraced-cron retraced-nsqd retraced-postgres
   ```

-> **Note**: Some of the `docker stop` commands may return “Container not found” errors because not every Replicated install has every container.

#### Step 7: Validate migration success

Finally, test that your new Terraform Enterprise installation works properly.

If you have an existing suite of release acceptance tests, execute those instead of doing the following steps.

> We recommend testing capabilities that you use in production, as you would for a Terraform Enterprise upgrade. For example: If you are using sentinel or run tasks in production, we recommend testing a run that includes these integrations in a lower environment before deploying to production.

Author: This one seems duplicate

Upgrade your Replicated-hosted Terraform Enterprise installation to [v202309-1](/terraform/enterprise/releases/2023/v202309-1) or later. Refer to [Upgrading](/terraform/enterprise/deploy/replicated/administration/infrastructure/upgrades)
for instructions.

#### Step 3: Prepare the custom Helm Values file for Terraform Enterprise

1. On Terraform Enterprise (Replicated), view existing configuration:

   ```shell-session
   $ replicatedctl app-config export
   ```

> 1. Create a custom [Helm Values file](https://helm.sh/docs/chart_template_guide/values_files/) e.g., `overrides.yaml` to override the [default values](https://github.com/hashicorp/terraform-enterprise-helm/blob/main/values.yaml) in the [Terraform Enterprise Helm chart](https://github.com/hashicorp/terraform-enterprise-helm/tree/main).

Reviewer: ```suggestion
1. Create a custom [Helm Values file](https://helm.sh/docs/chart_template_guide/values_files/), for example, `overrides.yaml` to override the [default values](https://github.com/hashicorp/terraform-enterprise-helm/blob/main/values.yaml) in the [Terraform Enterprise Helm chart](https://github.com/hashicorp/terraform-enterprise-helm/tree/main).
```
Per the [style guide](https://github.com/hashicorp/web-unified-docs/blob/main/docs/style-guide/general/language.md#do-not-use-words-or-phrases-borrowed-from-other-languages-scientific-words-or-jargon-words)

#### Step 3: Validate migration success

Finally, test that your new Terraform Enterprise installation works properly.

If you have an existing suite of release acceptance tests, you can use those instead of doing the following steps.

> We recommend testing capabilities that you use in production, as you would for a Terraform Enterprise upgrade. For example: If you are using sentinel or run tasks in production, we recommend testing a run that includes these integrations in a lower environment before deploying to production.

Reviewer: ```suggestion
We recommend testing capabilities that you use in production, just as you would for a Terraform Enterprise upgrade. For example, if you are using Sentinel or run tasks in production, we recommend testing a run that includes these integrations in a non-production environment before deploying to production.
```

#### Step 6: Validate migration success

Complete the following steps to verify that your new Terraform Enterprise installation works as expected.

Alternatively, you can execute your existing suite of release acceptance tests.

> We recommend testing capabilities that you use in production, as you would for a Terraform Enterprise upgrade. For example: If you are using sentinel or run tasks in production, we recommend testing a run that includes these integrations in a lower environment before deploying to production.

Reviewer: ```suggestion
We recommend testing capabilities that you use in production, just as you would for a Terraform Enterprise upgrade. For example, if you are using Sentinel or run tasks in production, we recommend testing a run that includes these integrations in a non-production environment before deploying to production.
```

#### Step 5: Validate migration success

> Verify that your new Terraform Enterprise installation works properly. If you have an existing suite of release acceptance tests, you can use them instead of completing the following steps. We recommend testing capabilities that you use in production, as you would for a Terraform Enterprise upgrade. For example: If you are using sentinel or run tasks in production, we recommend testing a run that includes these integrations in a lower environment before deploying to production. You should be able to log in to your new Terraform Enterprise installation with the credentials previously used for your Replicated installation. 

Author: Applied your suggested paragraph breaks to separate the validation section into distinct logical ideas.
